src/main.rs

```diff
@@ -0,0 +1,46 @@
+#![recursion_limit = "1024"]
+
+#[macro_use]
+extern crate error_chain;
+extern crate glib;
+#[macro_use]
+extern crate log;
+
+use glib::{KeyFile, KeyFileFlags};
+
+const CONFIG_FILE_PATH: &str = "/etc/fedora-coreos-metrics/client.conf";
```

Comment:
I'd suggest to just reserve `/{run,etc,usr/lib}/fedora-coreos-metrics/config.d/` since the beginning and pick up config files from there.